grafana/grafana
# InfluxDB: annotations migration to backend

**What this PR does / why we need it**:
Part of https://github.com/grafana/grafana/issues/43076

This PR contains the changes needed to migrate annotations to the backend.

**Test scenarios**:

1. Run make devenv sources=influxdb,influxdb1,grafana grafana_version=8.2.4 (or other version before changes)
2. Open up http://localhost:3000 && http://127.0.0.1:3001 (to test older 8.2.4 version)
3. Select gdev-influxdb-influxql
    - create a dashboard
    - create an annotation query in your dashboard e.g. select * from logs where time >= now() - 30m and time <= now() (be careful with the - 30m above as setting a value too big will potentially return tens of thousands of results. Maybe a smaller time like - 10m to start and increment until you get a number of results you are happy with)
    - add a time series panel hooked up to influxdb
    - set time range to last 3 hours or so (at least enough to cover time range set on annotation query)
    - go back to your dashboard and toggle on the annotation query -> you should see annotations in your dashboard
    - go back to your panel in edit mode -> you should see annotations
4. Select gdev-influxdb1-influxql
    - Should behave like `gdev-influxdb-influxql`

PS: make sure 
[feature_toggles]
enable = influxdbBackendMigration
